Update Pricing table design to current version

gmjuhasz opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

* Changed design to match the current one here: TTzDk3dzct551hYWRXPFa3-fi-3908%3A48536
* Added option to show discount percentage
* Added option to add popover info to each item. This can be done by providing `tooltipInfo` and optoional `tooltipTitle` props to the PricingTabelItem component.
* Switched up prices, now the current one is on the left.
* Added Terms of Service note to the bottom of the table.
* Now only the main column should have a white background, this can set by adding the `primary` prop to a PricingTableColumn.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

1202844140176868-as-1202944779121698

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

No.

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Fire up storybook with running `npm run storybook:dev` from the `js-packages/storybook` folder.
* Play with the added options described in the proposed changes section
* Check with Figma file if design matches.

gmjuhasz committed:
Update design based on Figma file

gmjuhasz committed:
changelog

gmjuhasz committed:
Fix popover inside p

gmjuhasz committed:
Merge branch 'trunk' into update/pricing-table-design

gmjuhasz committed:
Merge branch 'trunk' into update/pricing-table-design

gmjuhasz committed:
Add ToS and @wordpress/elements package

danielpost commented on projects/js-packages/components/components/icon-tooltip/index.tsx:
```diff
@@ -57,14 +58,11 @@ const IconTooltip: React.FC< IconTooltipProps > = ( {
 		placement,
 		animate,
 		noArrow: false,
-		resize: false,
-		flip: false,
```
I don't think these can be deleted. They're getting passed to [the Popover component](https://wordpress.github.io/gutenberg/?path=/story/components-popover--default) further down in the component.

dognose24 replied:
Yes! They are intentionally set to override the default value to the `Popover` component, which keeps the panel displaying fixed for usage along with the icon.

gmjuhasz replied:
Yeah, I deleted these because they caused some JS errors in the console:

<img width="914" alt="CleanShot 2022-09-20 at 09 42 05@2x" src="https://user-images.githubusercontent.com/36671565/191197377-39eec819-2c07-4365-b896-5d6901de9ac4.png">

I followed the Popover declaration to the interface, and didn't see these attributes there, so I thought they changed, but I was wrong then sorry!

What do you think about these errors?

danielpost replied:
It looks like that's this error: https://styled-components.com/docs/faqs#why-am-i-getting-html-attribute-warnings

I can reproduce it. Those props were [only added recently](https://github.com/WordPress/gutenberg/commit/63d89fb7f5973aa7afb617ba33b85d16788aceaf) to the `Popover` component.

It appears to be a mismatch between the latest version of the component and the version of `@wordpress/components` we're using in the monorepo. These props don't exist on the component yet in the version we're using, so they're get passed directly to the underlying HTML element (something like `<div flip={ false } resize={ false }>`.

It's probably fine to just ignore this warning for now, it'll go away when the dependency gets updated.

danielpost commented on projects/js-packages/components/components/pricing-table/index.tsx:
```diff
@@ -49,20 +71,34 @@ export const PricingTableItem: React.FC< PricingTableItemProps > = ( {
+			<Text variant="body-small">{ label || defaultLabel }</Text>
+			{ tooltipInfo && (
+				<IconTooltip
+					title={ tooltipTitle }
```
This can be `undefined`, but I guess that's not a problem since React just ignored `undefined` props right?

gmjuhasz replied:
Yeah, I am pretty sure that's the case, but I will give this another test to make double sure!

danielpost commented on projects/js-packages/components/components/product-price/index.tsx:
```diff
@@ -26,9 +27,17 @@ const ProductPrice: React.FC< ProductPriceProps > = ( {
 
 	showNotOffPrice = showNotOffPrice && offPrice != null;
 
+	const offPercentage = 100 - Math.round( ( offPrice / price ) * 100 );
```
What do you think about making this something more generic like `label` (instead of `showOffPercentage`? That way we could still use it for things like 50% off but also for stuff like `NEW` or `PROMO`.

danielpost reviewed (commented):
Really nice work! I've only added a couple minor thoughts.

Two other things:

1) It seems like the non-functional tooltip icons are still present on the table items. I think these need to be replaced by working tooltips, and on mobile these tooltips need to be present on each item in a column.

<img width="365" alt="Screenshot 2022-09-20 at 00 02 41@2x" src="https://user-images.githubusercontent.com/1713699/191127694-594c12d6-e50d-425a-b58d-16799513372c.png">

2) This was a bug that was already present before your PR, but adding a price without a label causes the button to line up wrong. What do you think about fixing that in this PR as well?

<img width="796" alt="image" src="https://user-images.githubusercontent.com/1713699/191127901-6bedda76-5859-48ec-9450-2dc0e5440313.png">

gmjuhasz commented:
Thanks @danielpost! I have a few questions about these:

> 1. It seems like the non-functional tooltip icons are still present on the table items. I think these need to be replaced by working tooltips

Sure, these can be changed as well! Should these be added by users, or should they have a default value?  Should they always appear on desktop? I was just wondering if a straight-forward feature name don't need one, we might should not show anything there.

> , and on mobile these tooltips need to be present on each item in a column.

Now my solution accepts one label for each table item, which is shown on large screens and on mobile as well. I guess if you want different popovers on mobile/desktop, or to have one on mobile but not on desktop, you can do that with using the `tooltipInfo={ isLg ? null : 'Only mobile label' } />` approach. Do you think this is good enough, or do you have a better solution? If needed I can create another prop for mobile tooltips, which overwrites the default one on mobile, could be cleaner.

> This was a bug that was already present before your PR, but adding a price without a label causes the button to line up wrong. What do you think about fixing that in this PR as well?

I'll look into this for sure, thanks for mentioning! 👍

danielpost commented:
> Should these be added by users, or should they have a default value?

Added by users IMO. They provide extra information about a specific row in the table, so I don't know if there's a default that would make sense.

> Should they always appear on desktop?

Here's how I think it's supposed to work: Each row in the table can have a tooltip. On desktop, these appear next to the "title" of each row, and on mobile they appear next to each related item in a column (so if you have two columns, the same tooltip would appear in two different places).

Additionally, users can add the tooltip to an individual item (you already added this feature). I think on mobile these would override the tooltips for a row, but that's something we can double-check with Design.

Does that make sense?

gmjuhasz committed:
Address review
      - Fix bug with height with no leyend on price
      - Add popover for feature on mobile
        - These are overwritten if user gives specific info for an item
      - Replaced props for Popover component
      - made offPercentage to a promoLabel to be able to display other values

gmjuhasz committed:
Merge branch 'trunk' into update/pricing-table-design

gmjuhasz committed:
Fix tests

gmjuhasz committed:
Fix popover clip-path issue

gmjuhasz committed:
Merge branch 'trunk' into update/pricing-table-design

danielpost commented on projects/js-packages/components/components/pricing-table/index.tsx:
```diff
@@ -51,9 +51,13 @@ export const PricingTableItem: React.FC< PricingTableItemProps > = ( {
+	const defaultTooltipTitle = items[ index ].tooltipTitle;
 	const includedLabel = __( 'Included', 'jetpack' );
 	const notIncludedLabel = __( 'Not included', 'jetpack' );
+	const showTooltip = tooltipInfo || ! isLg;
```
Do we need to add a check here to make sure either `tooltipInfo` or the `defaultTooltipInfo` prop is available, before we show it on mobile?

gmjuhasz replied:
If we treat the defaultTooltip optional then yes, I added it here too

danielpost commented on projects/js-packages/components/components/pricing-table/index.tsx:
```diff
@@ -97,17 +137,33 @@ const PricingTable: React.FC< PricingTableProps > = ( { title, items, children }
+								<Text variant="body-small">
+									<strong>{ item.name }</strong>
+								</Text>
+								<IconTooltip
+									title={ item.tooltipTitle }
+									iconClassName={ styles[ 'popover-icon' ] }
+									className={ styles.popover }
+									placement={ 'bottom-end' }
+									iconSize={ 22 }
+								>
+									<Text>{ item.tooltipInfo }</Text>
+								</IconTooltip>
```
Is `item.tooltipInfo` required? I think it should be optional, in which case we might need to add an extra check here.

gmjuhasz replied:
Yeah, I changed to optional in the end but the check here is right too!

danielpost commented on projects/js-packages/components/components/pricing-table/styles.module.scss:
```diff
@@ -25,14 +27,16 @@
+
+			&:nth-child(2) {
+				box-shadow: 0 4px 4px rgb(0 0 0 / 5%);
+			}
```
I know I suggested this change, but looking at it some more there's some shadow coming out now from the edges:

![d6fb1db2adf913d3111ce1ee67128401f760de38b7748f65b7a5b3fb874defa6](https://user-images.githubusercontent.com/1713699/191275539-a7afb9b9-c2fd-428e-ab9c-1613a82f50ac.png)

I think we can fix it like this:

```scss
&.is-primary {
	> * {
		background: var( --jp-white );
		position: relative;

		&::after {
			content: '';
			position: absolute;
			top: 0;
			left: 0;
			right: 0;
			bottom: 0;
			z-index: -1;
			box-shadow: 0px 4px 24px rgba( 0, 0, 0, 0.05 );
		}
	}
}
```

There's also a `clip-path: none` a couple lines down that's now obsolete.

danielpost commented on projects/js-packages/components/components/pricing-table/styles.module.scss:
```diff
@@ -69,6 +73,10 @@
 			left: var( --border-offset, 0 );
 			right: var( --border-offset, 0 );
```
These two lines (as well as the `--border-offset` variable declared earlier) probably don't make sense anymore right?

danielpost commented on projects/js-packages/components/components/pricing-table/styles.module.scss:
```diff
@@ -101,3 +109,42 @@
+}
+
+.tos {
+	padding-left: var( --padding );
+	padding-right: var( --padding );
```
Does this padding make sense on mobile?

gmjuhasz replied:
Actually, it might not need it, I removed it!

danielpost reviewed (commented):
Nice! Just a couple more thoughts. Did you look into the button positioning when the price doesn't have a label underneath it?

gmjuhasz committed:
Address review#2

gmjuhasz committed:
Merge branch 'trunk' into update/pricing-table-design

gmjuhasz commented:
> Did you look into the button positioning when the price doesn't have a label underneath it?

Yeah, the pseudo element [here](https://github.com/Automattic/jetpack/pull/26253/files#diff-a882645f666132f26df808f1d470641ecf4ed180c89b5e75f4a66310e05d750d) solves it!

oskosk committed:
Merge branch 'trunk' into update/pricing-table-design

oskosk committed:
Merge branch 'trunk' into update/pricing-table-design

oskosk committed:
Update versions

oskosk commented:
Pushed rebase in [e218a8f](https://github.com/Automattic/jetpack/pull/26253/commits/e218a8f566e84198c8621b2377f4536bf3b79d26) and versions update in [66015b5](https://github.com/Automattic/jetpack/pull/26253/commits/66015b573c4c9599918ae0ebcd1b30f7eee2f5bc)

oskosk reviewed (approved):
LGTM! Merging. Please follow up with another PR if something looks off or any feedback was left unaddressed.

oskosk commented:
r252976-wpcom
